lib/backend/pgbk/common/utils.go

```diff
@@ -131,7 +131,12 @@ func retry[T any](ctx context.Context, log logrus.FieldLogger, isIdempotent bool
 		var pgErr *pgconn.PgError
 		_ = errors.As(err, &pgErr)
 
-		if pgErr != nil && (pgErr.Code == pgerrcode.SerializationFailure || pgErr.Code == pgerrcode.DeadlockDetected) {
+		if pgErr != nil &&
+			(pgErr.Code == pgerrcode.SerializationFailure ||
+				pgErr.Code == pgerrcode.DeadlockDetected ||
+				pgErr.Code == pgerrcode.UniqueViolation ||
+				pgErr.Code == pgerrcode.ExclusionViolation) {
```

Comment:
Can you move this condition to a function and give it a name? `IsViolentError()`?

Reply:
Extracted in `isSerializationErrorCode` in 37a971c166